### PR TITLE
testing: extend transactions signature verification testing

### PR DESCRIPTION
**Pull request by algoidan**
## Summary

Add several unit tests for verifying types of signatures on transactions. including: regular signature, multisignature and logicsignature

### CODE REVIEW

File: data/transactions/verify/txn_test.go [hunk 1]
```diff
@@ -460,14 +663,176 @@ func TestTxnGroupCacheUpdate(t *testing.T) {
...
+				LastValid:   basics.Round(100),
+				GenesisHash: crypto.Hash([]byte{1, 2, 3, 4, 5}),
+			},
+			PaymentTxnFields: transactions.PaymentTxnFields{
+				Receiver: multiAddress[r],
+				Amount:   basics.MicroAlgos{Raw: uint64(a)},
+			},
+		}
+
+		op, err := logic.AssembleString(`arg 0
```
Comment (algorandskiy) [inline data/transactions/verify/txn_test.go:785]: please comment the point of having non-trivial program here

**Review by algorandskiy — COMMENTED** [review]
Looks reasonable, but please explain the logic sig program in the last test

File: data/transactions/verify/txn_test.go (hunk 1 above)
Comment (algorandskiy) [reply data/transactions/verify/txn_test.go:792]: I see, it checks it with valid and broken signature.
Maybe add a comment to `TestTxnGroupCacheUpdateLogicWithMultiSig` just to explain the test idea/scenario

File: data/transactions/verify/txn_test.go [hunk 2]
```diff
@@ -60,25 +61,122 @@ func verifyTxn(s *transactions.SignedTxn, txnIdx int, groupCtx *GroupContext) er
 	return batchVerifier.Verify()
 }
 
+type DummyLedgerForSignature struct {
+}
+
+func (d *DummyLedgerForSignature) BlockHdrCached(basics.Round) (bookkeeping.BlockHeader, error) {
+	return bookkeeping.BlockHeader{
```
Comment (algonautshant) [inline data/transactions/verify/txn_test.go:68]: `bookkeeping.BlockHeader` creation is repeated more than 5 times.
Will be nice to make a function that returns a block header, and use the function everywhere.

**Review by algonautshant — APPROVED** [review]
Look Great!
This is very nicely structured, and can easily adapted for other tests.
Just one minor comment.

**Review by algonautshant — APPROVED** [review]

**Review by algorandskiy — APPROVED** [review]

**Pull request merged**